### PR TITLE
CDAP-12769 Use runid in the directory name.

### PR DESCRIPTION
**Pull request by sagarkapare**
JIRA - https://issues.cask.co/browse/CDAP-12769

### CODE REVIEW

**Review by anwar6953 — APPROVED** [review]

File: cdap-spark-core-base/src/main/scala/co/cask/cdap/app/runtime/spark/AbstractSparkExecutionContext.scala [hunk 1]
```diff
@@ -215,7 +215,7 @@ abstract class AbstractSparkExecutionContext(sparkClassLoader: SparkClassLoader,
     val cConf = runtimeContext.getCConfiguration
     val tempDir = new File(cConf.get(Constants.CFG_LOCAL_DATA_DIR),
                            cConf.get(Constants.AppFabric.TEMP_DIR)).getAbsoluteFile
-    val classDir = new File(tempDir, runtimeContext.getProgramRunId + "-classes-" + interpreterCount.incrementAndGet())
+    val classDir = new File(tempDir, runtimeContext.getProgramRunId.getRun + "-classes-" + interpreterCount.incrementAndGet())
```
Comment (chtyim) [inline cdap-spark-core-base/src/main/scala/co/cask/cdap/app/runtime/spark/AbstractSparkExecutionContext.scala]: This makes the directory a bit hard to relate to. Probably better use the program run id and form a string that is safe for all FS. Maybe just use `.` to separate out the parts.

Comment (chtyim) [reply cdap-spark-core-base/src/main/scala/co/cask/cdap/app/runtime/spark/AbstractSparkExecutionContext.scala]: Specifically, if you turn the `toIdParts` method in `EntityId` to public, then you can simply do `runtimeContext.getProgramRunId.toIdParts.mkString(".")` in here.

**Comment by chtyim** [comment]
LGTM. Make sure it is tested on Windows.

**Comment by sagarkapare** [comment]
Build running here - https://builds.cask.co/browse/CDAP-RUT1450-1

**Comment by sagarkapare** [comment]
Build passed. Squashing and merging.

**Pull request merged**